StyleCop.Analyzers/StyleCopTester/Program.cs

```diff
@@ -79,7 +81,115 @@ private static void Main(string[] args)
+
+            foreach (var codeFixer in codeFixers)
+            {
+                equivalenceGroups.AddRange(CodeFixEquivalenceGroup.CreateAsync(codeFixer, diagnostics, solution).GetAwaiter().GetResult());
+            }
+
+            Console.WriteLine($"Found {equivalenceGroups.Count} equivalence groups.");
+
+            Console.WriteLine("Calculating changes");
+
+            foreach (var fix in equivalenceGroups)
+            {
+                try
+                {
+                    stopwatch.Restart();
+                    Console.WriteLine($"Calculating fix for {fix.CodeFixEquivalenceKey}.");
+                    fix.GetOperationsAsync(CancellationToken.None).GetAwaiter().GetResult();
+                    WriteLine($"Calculating changes completed in {stopwatch.ElapsedMilliseconds}ms", ConsoleColor.Yellow);
+                }
+                catch (Exception ex)
+                {
+                    // Report thrown exceptions
+                    WriteLine($"The fix {fix.CodeFixEquivalenceKey} threw an exception:", ConsoleColor.Yellow);
+                    WriteLine(ex.ToString(), ConsoleColor.Yellow);
+                }
+            }
+        }
+
+        private static void WriteLine(string text, ConsoleColor color)
+        {
+            Console.ForegroundColor = color;
+            Console.WriteLine(text);
+            Console.ResetColor();
+        }
+
+        private static void TestCodeFixes(Stopwatch stopwatch, Solution solution, ImmutableList<Diagnostic> diagnostics)
+        {
+            Console.WriteLine("Calculating fixes");
+
+            List<CodeAction> fixes = new List<CodeAction>();
+
+            var codeFixers = GetAllCodeFixers();
+
+            foreach (var item in diagnostics)
+            {
+                foreach (var codeFixer in codeFixers.GetValueOrDefault(item.Id, ImmutableList.Create<CodeFixProvider>()))
+                {
+                    fixes.AddRange(GetFixesAsync(solution, codeFixer, item).GetAwaiter().GetResult());
+                }
             }
+
+            Console.WriteLine($"Found {fixes.Count} potential code fixes");
+
+            Console.WriteLine("Calculating changes");
+
+            stopwatch.Restart();
+
+            object lockObject = new object();
+
+            Parallel.ForEach(fixes, fix =>
+            {
+                try
+                {
+                    fix.GetOperationsAsync(CancellationToken.None).GetAwaiter().GetResult();
+                }
+                catch (Exception ex)
+                {
+                    // Report thrown exceptions
+                    lock (lockObject)
+                    {
+                        Console.ForegroundColor = ConsoleColor.Yellow;
+                        Console.WriteLine($"The fix '{fix.Title} 'threw an exception:");
+                        Console.ForegroundColor = ConsoleColor.Red;
+                        Console.WriteLine(ex);
+                        Console.ResetColor();
```

Comment:
Why manually set the color and call reset color here, yet on lines 126-127 you call the convenience method?